### DISCUSSION TITLE
Empty rawAdvertisement of BleClient.requestLEScan

### DISCUSSION THREAD

**Discussion by PraveenNale** [capacitor-community/bluetooth-le#525] [Q&A] [answered]
Hi,
Facing issues with receiving ble advertisements after recent app (ionic-react) upgrade - result.rawAdvertisement of BleClient.requestLEScan is empty. nRFConnect app on same android phone as well as btmon on linux is able to receive the advertisements [size 23B]. Even earlier version of app received the advertisements fine. Unfortunately I don't have a backup of the working app or the version of libraries being used earlier. Tried downgrading @capacitor* libraries but that didn't help. Would appreciate any pointers on how to proceed/debug further...

Thanks,
Praveen.

Code snippet to print requestLEScan results:
```
await BleClient.requestLEScan({
  namePrefix: "INDZ",
  allowDuplicates: true,
  scanMode: 2,
}, (result) => {
  setMessage("Adv:" + JSON.stringify(result));
});
```

Received advertisement result:
```
{
  device: {
    deviceId: "AB:CD:EF:12:34:56",
    name: "INDZABCDEF"
  }
  localName: "INDZABCDEF",
  rssi: -58,
  txPower: 127,
  manufacturerData: "",
  serviceData: {
    0000181c-0000-1000-8000-00805f9b34fb: {}
  },
  uuids: [],
  rawAdvertisement: {}
}
```

Build environment:
![package json](https://user-images.githubusercontent.com/130972469/232538581-f8c15cc2-2473-408c-b6e6-bc37ef12cc84.png)

Devices tested on: Android 8.1 & 12, Android Chrome with flags enabled

**Comment by pwespi** [comment, marked as answer]
> Devices tested on: Android 8.1 & 12, Android Chrome with flags enabled

If your testing in the Chrome browser on Android (or any other OS), `rawAdvertisement` will always be empty, because in the Chrome browser the web version of the plugin is used. Only when building the app as a native Android Capacitor app `rawAdvertisement` will be set.

If you're building a Android Capacitor app, then `JSON.stringify` is not a good method to inspect the `rawAdvertisement`, because it is a `DataView` and `JSON.stringify(new DataView(Uint8Array.from([1, 2]).buffer))` returns `{}`. If you want to transform the DataView into an array of numbers, there is the `dataViewToNumbers` helper method.

**Comment by PraveenNale** [comment]
Thanks!
JSON.stringify(result) was the culprit, JSON.stringify(dataViewToNumbers(result.rawAdvertisement)) shows the advertisements.
I've uploaded the [app](https://github.com/PraveenNale/bleAds) just in case someone needs to experiment with requestLEScan in Ionic.
